OAK-11979: Clean up *StringValue in JCR XML import

reschke opened the pull request.

reschke committed:
OAK-11979: Clean up *StringValue in JCR XML import

reschke commented:
Seems we have mo coverage for Base64 encoded binaries.

rishabhdaim reviewed (commented):
Can we create unit cases for these files, just to compare the output before & after.

thomasmueller committed:
OAK-11977 Tree store: BufferOverflowException (#2571)

waldoro committed:
OAK-11899 - use default value in case config value can not be parsed (#2509)

* OAK-11899 - use default value in case config value can not be parsed

* OAK-11899 - if config parameter can not be converted, use default value

* OAK-11899 - increase org.apache.jackrabbit.oak.spi.security version

mbaedke commented on oak-jcr/src/main/java/org/apache/jackrabbit/oak/jcr/xml/StringValue.java:
```diff
@@ -63,12 +61,11 @@ public String getString() {
-            ByteArrayOutputStream baos = new ByteArrayOutputStream();
             try {
-                Base64.decode(value, baos);
+                byte[] bytes = Base64.getDecoder().decode(value);
```
Maybe inline that?

reschke replied:
inline?

joerghoh committed:
OAK-11969 do not check the existence of the tree twice (#2560)

reschke commented:
Before we proceed here, we need test coverage. Optimally separate ticket / PR.

jsedding committed:
OAK-11934 - segment preloading for PersistentCache (#2569)

rishabhdaim committed:
OAK-11914 : removed usage of Guava's HashBasedTable (#2573)

* OAK-11914 : removed usage of Guava's HashBasedTable

* OAK-11914 : removed un-necessary usage of computeIfAbsent method

rishabhdaim committed:
OAK-11911 : exposed DirectExecutor from ExecutorUtils (#2575)

* OAK-11910 : exposed DirectExecutor from ExecutorUtils

* OAK-11910 : marked DirectExecutor as public again

* OAK-11911 : increased minor version

reschke committed:
OAK-11981: blob-plugins - remove use of TransientFileFactory (#2576)

reschke committed:
OAK-11983: benchmarks: remove jackrabbit-core dependency (#2577)

jsedding committed:
OAK-11980 - improve parallelization of I/O during segment-azure initialization (#2574)

rishabhdaim committed:
OAK-11985 : added MongoServerUnavailableException into transient errors (#2582)

joerghoh committed:
OAK-11982 move the calculation of the WARN limit out of the constructor (#2585)

rishabhdaim committed:
OAK-11913 : created Forwarding executir service (#2580)

rishabhdaim committed:
OAK-11935 : updating aws sdk from 1.x to 2.x (#2558)

* OAK-11935 : updating aws sdk from 1.x to 2.x

* OAK-11935 : added more unit cases

* OAK-11935 : added unit cases for S3RequestDecorator

* OAK-11935 : fixed issue when running ITs with different encryption mode

* OAK-11935 : removed unused imports

* OAK-11935 : incorporated review comments

* OAK-11935 : added review comments

* OAK-11935 : added properties for cross Region access

* OAK-11935 : removed additional list for removing keys in case of GCP mode

* OAK-11935 : moved out CRUD operations from Utils to S3CrudHelper

* OAK-11935 : added unit cases for waitForBucket method

* OAK-11935 : make total attemps and delay configurable to use lower values in unit cases

* OAK-11935 : remaned S3CrudHelper to S3BackendHelper

johobot committed:
OAK-11987 - org.apache.jackrabbit.oak.segment.azure.tool.SegmentStoreMigrator.migrateBinaryRef is missing a null check (#2586)

nscendoni committed:
OAK-11984 Support UserId Change for External Users (#2581)

* OAK-11984 Support UserId Change for External Users

* Removed unused change

* Update oak-auth-external/src/main/java/org/apache/jackrabbit/oak/spi/security/authentication/external/impl/ExternalLoginModule.java

Co-authored-by: Alejandro Moratinos <Amoratinos@users.noreply.github.com>

* Update oak-auth-external/src/main/java/org/apache/jackrabbit/oak/spi/security/authentication/external/impl/ExternalLoginModule.java

Co-authored-by: Alejandro Moratinos <Amoratinos@users.noreply.github.com>

* moving constants

* Added FF

* Added tests for FF

* Added debug log

---------

Co-authored-by: Alejandro Moratinos <Amoratinos@users.noreply.github.com>
Co-authored-by: angela <anchela@adobe.com>

ChlineSaurus committed:
OAK-11936: Allow updating the inference config via JMX (#2525)

* OAK-11936: Allow updating the inference config via JMX

* OAK-11936: Apache License added to new classes

---------

Co-authored-by: marvinw <marvinw@adobe.com>

ChlineSaurus committed:
OAK-11949: Sort union queries without "order-by" by score (#2540)

* OAK-11949: Sort Union Queries without order-by by score

* OAK-11936: restructure if-clause, move import and one additional test

* OAK-11949: use double for Null protection

---------

Co-authored-by: marvinw <marvinw@adobe.com>

rishabhdaim committed:
OAK-11912 : created DirectExecutorServiuce in oak-commons (#2579)

reschke committed:
OAK-11936: Allow updating the inference config via JMX - fix package version

reschke committed:
OAK-11936: Allow updating the inference config via JMX - line ends in test case

reschke committed:
OAK-11994 remove unused dependency to joda-time (#2593)

Co-authored-by: Joerg Hoh <jhoh@adobe.com>

Joscorbe committed:
OAK-11997: Log slow Mongo queries in DocumentNodeStore (#2596)

reschke committed:
Revert "OAK-11936: Allow updating the inference config via JMX - line ends in test case"

This reverts commit dea095688e37c9a5c0a99694cc8d0d42f2de831a.

reschke committed:
Revert "OAK-11936: Allow updating the inference config via JMX - fix package version"

This reverts commit 7081c97e897f036211ad679eb6bd6725d2fb89aa.

reschke committed:
Revert "OAK-11936: Allow updating the inference config via JMX (#2525)"

This reverts commit 174dce112b33c0f3e59f665f96ab3ed390397506.

reschke committed:
Reapply "OAK-11936: Allow updating the inference config via JMX (#2525)"

This reverts commit acac0de477b4fe32e3a3268dd12bbef72e9267af.

reschke committed:
Reapply "OAK-11936: Allow updating the inference config via JMX - fix package version"

This reverts commit 248d4ec344d33c2fff6b90a13874c72cccfc10d4.

reschke committed:
Reapply "OAK-11936: Allow updating the inference config via JMX - line ends in test case"

This reverts commit 09a603ba0bac80f5907902b36443f158a21f8aab.

reschke committed:
OAK-11996: Expand oak/docs/participating wrt to test coverage and Sonar (#2594)

reschke committed:
OAK-11999: Release Oak 1.88.0 - Candidate Release Notes

reschke committed:
[maven-release-plugin] prepare release jackrabbit-oak-1.88.0

reschke committed:
[maven-release-plugin] prepare for next development iteration

fabriziofortino committed:
OAK-11998: bump ES to 8.19.5, fix k for knn queries (#2597)

* OAK-11998: bump ES to 8.19.5, fix k for knn queries

* OAK-11998: fix test

* OAK-11998: add comment

* OAK-11998: fix lucene version

fabriziofortino committed:
OAK-11898: decouple ElasticSecureFacetAsyncProvider from async iterator (#2564)

* OAK-11899: decouple ElasticSecureFacetAsyncProvider from async iterator

* Update oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/async/facets/ElasticSecureFacetAsyncProvider.java

Co-authored-by: Thomas Mueller <thomas.tom.mueller@gmail.com>

* OAK-11899: fix error, improved logs

* OAK-11899: fix log message

* OAK-11899: (minor) comment

* OAK-11899: improved log message

* OAK-11899: add missing exception stacktrace

* OAK-11998: make size configurable and add check based on limitReads

---------

Co-authored-by: Thomas Mueller <thomas.tom.mueller@gmail.com>

reschke committed:
OAK-11999: Release Oak 1.88.0 - Candidate Release Notes

reschke committed:
[maven-release-plugin] prepare release jackrabbit-oak-1.90.0

reschke committed:
[maven-release-plugin] prepare for next development iteration

reschke committed:
Revert "[maven-release-plugin] prepare for next development iteration"

This reverts commit ce81d652ba3eed76d5c83f635b438d4961280159.

reschke committed:
Revert "[maven-release-plugin] prepare release jackrabbit-oak-1.90.0"

This reverts commit 0c237039e7514f999d4806e39cb14f7b7e6d2d6a.

reschke committed:
[maven-release-plugin] prepare release jackrabbit-oak-1.88.0

reschke committed:
[maven-release-plugin] prepare for next development iteration

reschke committed:
OAK-12001: run CI on branch 1.22 as well (#2602)

reschke committed:
OAK-11999: Release Oak 1.88.0 - update oak-doc* version numbers

ierandra committed:
OAK-12003 Refactor code in AzurePersistenceManager (#2606)

ierandra committed:
OAK-11991 Optimize the oak-segment recovery process

thomasmueller committed:
OAK-12004 Datastore: speedup datastore copy (#2607)

* OAK-12004 Datastore: speedup datastore copy

* OAK-12004 Datastore: speedup datastore copy

* OAK-12004 Datastore: speedup datastore copy

* OAK-12004 Datastore: speedup datastore copy

jsedding committed:
OAK-12005 - segment preloading graph-cache uses too much heap (#2611)

Joscorbe committed:
OAK-12011: add sleep to avoid backgroundPurge to interfere with tests (#2617)

thomasmueller committed:
OAK-12012 Sporadic SessionSaveDelayerConfigTest failure (#2619)

rishabhdaim committed:
OAK-12008 : Embed aws sdk 2.x jars into oak-blob-cloud bundle (#2616)

reschke committed:
OAK-12015: Improve diagnostics on property related constraint violations

rishabhdaim committed:
OAK-11916 : added exiting executor service in oak-commons (#2622)

* OAK-11916 : added exiting executor service in oak-commons

* OAK-11916 : updated package version

* OAK-11916 : fixed failing issues due to wrong guava import

rishabhdaim committed:
OAK-11917 : added exiting scheduled executor service in oak-commons (#2623)

mbaedke committed:
OAK-11617: Provide oak-run commands to analyze and fix inconsistencie… (#2615)

* OAK-11617: Provide oak-run commands to analyze and fix inconsistencies in the namespace registry

Introduced new option --prune, added documentation and tests.

rishabhdaim committed:
OAK-11920 : removed usage of MoreExecutors.newDirectExecutorService with oak-commons implementation (#2624)

rishabhdaim committed:
OAK-11919 : removed usage of Guava's ListenableFutureTask from prod files (#2627)

reschke committed:
OAK-12021: cleanup oak-blob-plugins pom (oak-commons dependency)

reschke committed:
OAK-11830: Update pax-exam test dependency to 4.14.0

reschke committed:
OAK-11830: Update pax-exam test dependency to 4.14.0 - fix comp tests faling back to 4.13.1

reschke committed:
OAK-11830: Update pax-exam test dependency to 4.14.0 - exclude oak-store-compostie for now

thomasmueller committed:
OAK-12007 XPath with or conditions should not always be converted to union (#2613)

* OAK-12007 XPath with or conditions should not always be converted to union

* OAK-12007 XPath with or conditions should not always be converted to union

* OAK-12007 XPath with or conditions should not always be converted to union

* OAK-12007 XPath with or conditions should not always be converted to union

rishabhdaim committed:
OAK-12018 : replaced DirectExecutor.INSTANCE call with ExecutorUtils.directExecutor (#2628)

rishabhdaim committed:
OAK-12019 : replaced Guava's MoreExecutors.getExitingExecutorService with oak-commons implementation (#2634)

rishabhdaim committed:
OAK-12020 : replaced Guava's MoreExecutors.getExitingScheduledExecutorService with oak-commons implementation (#2633)

rishabhdaim committed:
OAK-12009 : provided support for GCP for new AWS sdk 2.x (#2621)

* OAK-12009 : provided support for GCP for new AWS sdk 2.x

* OAK-12009 : ircorporated review comments

* OAK-12009 : avoid reading files into memory

* OAK-12009 : extracted out getting length from InputStream to a separate method

rishabhdaim committed:
OAK-12025 : replaced Guava's SameThreadExecutorService (#2637)

rishabhdaim committed:
OAK-11918 : removed Guava's ListeningExecutorService (#2638)

reschke committed:
OAK-11731: expose NodeCounter (getEstimatedChildNodeCounts) in Jackra… (#2625)

rishabhdaim committed:
OAK-12027 : added api to remove node and its descendants starting fro… (#2636)

* OAK-12027 : added api to remove node and its descendants starting from leaf nodes upto parents

* OAK-12027 : improved comments

rishabhdaim committed:
OAK-11924 : replaced Guava's ForwardingListeningExecutorService with … (#2640)

* OAK-11924 : replaced Guava's ForwardingListeningExecutorService with oak-commons implementation

* Update FileCacheTest.java

reschke committed:
OAK-11618: Remove usage of Guava Stopwatch (Groovy) (#2642)

Joscorbe committed:
OAK-10844: Disable BranchCommitGCTest due to flakiness (#2641)

rishabhdaim committed:
OAK-12029 : replaced Guava's settable future with CompletableFuture (#2644)

reschke committed:
OAK-11731: expose NodeCounter (getEstimatedChildNodeCounts) in JackrabbitNode - test case tuning (#2645)

rishabhdaim committed:
OAK-11923 : replaced Guava's ForwardingListenableFuture (#2643)

rishabhdaim committed:
OAK-11924 : replaced Guava's AbstractListeningExecutorService with native java implementation (#2646)

reschke committed:
OAK-11731: expose NodeCounter (getEstimatedChildNodeCounts) in JackrabbitNode - avoid 2 mbeans with the same name (#2648)

rishabhdaim committed:
OAK-11922 : removed usage of guava's MoreExecutors.listeningDecorator (#2647)

rishabhdaim committed:
OAK-12032 : replaced guava's UncheckedExecutionException with runtimeexception (#2650)

rishabhdaim committed:
OAK-12031 : added jackrabbit-jcr-commons since now jackrabbit-core is using it from its path used by oak-upgrade (#2649)

kwin committed:
Fix links

Remove linebreaks which destroyed markdown links

nfsantos committed:
OAK-12034 - CachingSegmentArchiveReader#containsSegment should not check the persistence cache for the segment (#2655)

nfsantos committed:
OAK-12036 - Expose stats in the PersistentCache (#2657)

rishabhdaim committed:
OAK-12033 : replaced Guava's Monitor with ReentrantLock (#2654)

thomasmueller committed:
OAK-12030 Upgrade LZ4 (#2662)

* OAK-12030 Upgrade LZ4

* OAK-12030 Upgrade LZ4

fabriziofortino committed:
OAK-12038: increase assertion time to avoid flakiness in ElasticFacetTest (#2661)

* OAK-12038: increase assertion time to avoid flakiness in ElasticFacetTest

* OAK-12038: increase assertion time to avoid flakiness in ElasticFacetTest

jsedding committed:
OAK-12040 - segment-azure: reduce HTTP requests for writes (#2665)

jsedding committed:
OAK-12041 - bump testcontainers to v1.21.4 (#2666)

smiroslav committed:
OAK-12039 Treat transient client-side exceptions as recoverable in AzureRepositoryLock (#2663)

* OAK-12039 Treat transient client-side exceptions as recoverable in AzureRepositoryLock

* OAK-12039 update verison of reactor-core

* OAK-12039 introduced oak.segment.azure.lock.leaseRenewalTimeoutInMs sys property

* OAK-12039 change the test to cause timeout exception instead of mocking it

* OAK-12039 retry on IllegalStateException

---------

Co-authored-by: smiroslav <miroslav@apache.com>

smiroslav committed:
OAK-12039 test that verifies that lease renewal in AzureRepositoryLockV8 correctly handles client timeout (#2667)

Co-authored-by: smiroslav <miroslav@apache.com>

nfsantos committed:
OAK-12043 - Avoid object allocation in SegmentID.compareTo (#2668)

nfsantos committed:
OAK-12044 - Optimize PathUtils.concat for the case where parent is the root directory. (#2669)

joerghoh committed:
preliminary release notes Oak 1.90.0

joerghoh committed:
[maven-release-plugin] prepare release jackrabbit-oak-1.90.0

joerghoh committed:
[maven-release-plugin] prepare for next development iteration

fabriziofortino committed:
OAK-12045: strip leading and trailing whitespaces (including newlines) from :fulltext (#2670)

mbaedke committed:
OAK-11960: oak-examples: The static content of the webapp still refers to RMI features that are no longer supported (#2678)

Fixed.

fabriziofortino committed:
OAK-12052 | oak-search-elastic: log entire request on query failure (#2680)

fabriziofortino committed:
OAK-12053 | oak-search-elastic: set max analyzed offset for highlights (#2681)

* OAK-12053: set max analyzed offset for highlights to avoid query failures for large fields

* OAK-12053: minor improvements

* OAK-12053: minor improvements

* OAK-12053: minor improvements

* OAK-12053: minor improvements

* OAK-12053: add TODO to improve logic when upgrading to v9.x

nfsantos committed:
OAK-12047 - CachingSegmentArchiveReader#readSegment should check if archive contains segment before looking up in the PersistentCache (#2672)

mbaedke committed:
OAK-12056: Speed up NamespaceRegistryCommandTest by reusing the NodeStore across tests. (#2683)

done

rishabhdaim committed:
OAK-12037 : remove usage of Guava's Monitor from oak-lucene (#2660)

reschke committed:
OAK-11224: remove guava dependencies from POMs (ongoing) - oak-lucene (#2685)

reschke committed:
OAK-12059: Update Oak trunk to Jackrabbit 2.22.3 (#2686)

rishabhdaim committed:
OAK-12061 : removed usage of guava's ThreadFactoryBuilder (#2691)

rishabhdaim committed:
OAK-12065 : added class to split locks similar to guava's Striped (#2692)

* OAK-12065 : added unit class to strip locks similar to guava's Striped

* OAK-12065 : incorporated review comments

rishabhdaim committed:
OAK-12062 : fixed NPE while uploading metadata for AWS S3 (#2688)

* OAK-12062 : fixed NPE while uploading metadata for AWS S3

* OAK-12062 : removed temp file in finally method

rishabhdaim committed:
OAK-12064 : replaced guava striped with oak-commons StripedLocks (#2694)

reschke committed:
OAK-12067: Utils.alignWithExternalRevisions logs timestamps with second resolution (#2693)

* OAK-12067: Utils.alignWithExternalRevisions logs timestamps with second resolution

* OAK-12067: Utils.alignWithExternalRevisions logs timestamps with second resolution

* OAK-12067: Utils.alignWithExternalRevisions logs timestamps with second resolution

jsedding committed:
OAK-12068 - segment graph UUID deduplication (OAK-12005) can be too inefficient (#2695)

- remove UUID deduplication in segment graphs
- avoid over-allocation of HashMaps of known bounded size

jsedding committed:
OAK-12069 - reduce SegmentPreloader memory usage (#2696)

- remove segment-graph cache
- read segment references from segment data

jsedding committed:
OAK-12070 - Reduce memory consumption of azure segment stores (#2699)

- deduplicate in-memory UUID instances (halves number of instances in memory
  at 32 bytes each)
- avoid LinkedHashMap in favour of java.util.ImmutableCollections (eliminates
  one LinkedHashMap$Entry weighing 40 bytes per segment)
- create GCGeneration instance pool (instances now used by RemoteSegmentArchiveEntry)
- use UUID and GCGeneration references in RemoteSegmentArchiveEntry (reduces
  object size from 48 to 32 bytes)

rishabhdaim committed:
OAK-11950 : removed guava's queues.drainUninterruptibly (#2702)

rishabhdaim committed:
OAK-12074 : added awaitUninterruptibly() in oak-commons (#2703)

* OAK-12074 : added awaitUninterruptibly() in oak-commons

* OAK-12074 : removed overloaded method, since it is not used

fabriziofortino committed:
OAK-12071: fix property removal for externally modifiable indexes (#2700)

* OAK-12071: fix property removal for externally modifiable indexes

* OAK-12071: fix flaky test increasing eventual assertion time

* OAK-12071: test improvements

* OAK-12071: ensure index name uniqueness in tests

* OAK-12071: fix case where properties are not correctly removed

* OAK-12071: minor code improvement

rishabhdaim committed:
OAK-12075 : replaced guava's Uninterruptible.awaitUninterruptibly with oak-commons (#2704)

rishabhdaim committed:
OAK-12076 : added sleepUninterruptibly() in oak-commons (#2705)

rishabhdaim committed:
OAK-12078 : replaced guava's Uninterruptible.sleepUninterruptibly with oak-commons (#2710)

rishabhdaim committed:
OAK-12081 : added isInOrder() in IterableUtils to check whether all element of given iterable are in order as specified by given comparator (#2713)

mbaedke committed:
OAK-10344: oak-run ITs fail with failsafe >= 3.0.0 (#2709)

Fixed.

rishabhdaim committed:
OAK-12077 : added joinUninterruptibly() in oak-commons (#2711)

* OAK-12077 : added joinUninterruptibly() in oak-commons

* OAK-12077 : added review comments

* OAK-12077 : updated sleep durations to sleep within joining period

* OAK-12077 : fixed assertion errors if occured in joiningThread

rishabhdaim committed:
OAK-12080 : replaced guava's Comparators.isInOrder with oak-commons implementation (#2715)

jsedding committed:
OAK-12069 - reduce SegmentPreloader memory usage (#2718)

- info log msg should only be logged on debug

reschke committed:
OAK-11941: Upgrade (shaded) Guava to 33.5.0

mbaedke committed:
OAK-10342: Update Apache Parent POM to version 37 (#2716)

done

rishabhdaim committed:
OAK-12083 : replaced guava's Uninterruptible.joinUninterruptibly with oak-commons (#2717)

bhabegger committed:
OAK-12057 - Select same plan with or without LIMIT option (#2684)

thomasmueller committed:
OAK-12057 - Select same plan with or without LIMIT option - version bump

thomasmueller committed:
Revert "OAK-12057 - Select same plan with or without LIMIT option - version bump"

This reverts commit 8a13d35b6e88ad959ec8d52a75a846f3bb02251e.

thomasmueller committed:
Revert "OAK-12057 - Select same plan with or without LIMIT option (#2684)"

This reverts commit 436f733f8a9208d21df29bbae8258b5774cc3aee.

bhabegger committed:
OAK-12054: Refactor creation of ThreadPoolExecutors (fix those that we in fact only running one thread) (#2679)

reschke committed:
Revert "OAK-12070 - Reduce memory consumption of azure segment stores (#2699)"

This reverts commit 082c3a3e551bd04663c785c6d78124ec415e784f.

bhabegger committed:
OAK-12063: Remove obsolete feature toggles FT_NAME_IMPROVED_IS_NULL_COST, FT_OPTIMIZE_IN_RESTRICTIONS_FOR_FUNCTIONS, and FT_NAME_PREFETCH_FOR_QUERIES (#2690)

jsedding committed:
OAK-12084 - Move GCGeneration class to exported SPI package (#2725)

reschke committed:
OAK-12086: oak-segment-tar: remove optional log4j dependency

jsedding committed:
OAK-12070 - Reduce memory consumption of azure segment stores (#2726)

- restores changes reverted in commit 9bfbce0b7dd8d457a5640dd97545f140931e7b44.

thomasmueller committed:
OAK-12010 Simplified index management (without optimizer) (#2689)

* OAK-12010 Simplified index management (without optimizer)

* OAK-12010 Simplified index management

* OAK-12010 Simplified index management

* OAK-12010 Simplified index management

* Update oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/diff/DiffIndex.java

Co-authored-by: Benjamin Habegger <benjamin@habegger.tech>

* Update oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/diff/DiffIndex.java

Co-authored-by: Benjamin Habegger <benjamin@habegger.tech>

* Update oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/diff/DiffIndex.java

Co-authored-by: Amit Jain <amitj@apache.org>

* Update oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/diff/DiffIndexMerger.java

Co-authored-by: Amit Jain <amitj@apache.org>

* Update oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/diff/DiffIndex.java

Co-authored-by: Amit Jain <amitj@apache.org>

* Update oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/diff/JsonNodeBuilder.java

Co-authored-by: Benjamin Habegger <benjamin@habegger.tech>

* Update oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/diff/DiffIndex.java

Co-authored-by: Amit Jain <amitj@apache.org>

* OAK-12010 Simplified index management

* OAK-12010 Simplified index management

* OAK-12010 Simplified index management

* OAK-12010 Simplified index management

* OAK-12010 Simplified index management

---------

Co-authored-by: Benjamin Habegger <benjamin@habegger.tech>
Co-authored-by: Amit Jain <amitj@apache.org>

rishabhdaim committed:
OAK-12087 : bump netty to 4.1.131.Final (#2731)

rishabhdaim committed:
OAK-12088 : added connectionAcquisitionTimeout to avoid acquire timeout issue (#2732)

reschke committed:
OAK-12082: add test coverage for org.apache.jackrabbit.oak.jcr.xml.BuffferedStringValue (#2728)

reschke committed:
OAK-12085: OutOfMemory in oak-segment-azure tests - reverting changes for OAK-12039 (#2738)

* Revert "OAK-12039 Treat transient client-side exceptions as recoverable in AzureRepositoryLock (#2663)"

This reverts commit e4a2b3916035a13adb0f63d686f3fd3ebaf47ef9.

* Revert "OAK-12039 test that verifies that lease renewal in AzureRepositoryLockV8 correctly handles client timeout (#2667)"

This reverts commit 6697d706d92f6928adc9768a3b8f2cb6f070f310.

thomasmueller committed:
OAK-12010 (Remove misplaced files)

reschke committed:
Revert "OAK-12085: OutOfMemory in oak-segment-azure tests - reverting changes for OAK-12039 (#2738)"

This reverts commit ef8f1d3d0dfee7f1e2d3c1a8817a00723d54a189.

jsedding committed:
OAK-12085 - OutOfMemory in oak-segment-azure tests (#2739)

- consistently close FileStores and other closeable objects

mbaedke committed:
OAK-12066: NPE in AstElement "quote()" (#2708)

Added diagnostic logging.

mbaedke committed:
OAK-11975: Use Mockito as an agent in Oak (#2653)

Replaced the dynamic inline mocking with a static agent dependency.

reschke committed:
OAK-12010: Simplified index management - fix JSON comparison in tests (#2741)

reschke committed:
OAK-12090: Update commons-lang3 dependency to 3.20.0 (#2736)

reschke committed:
OAK-12092: Update commons-text dependency to 1.15.0 (#2742)
